Reorganize (#3)
dalehamel/app_profiler: README.md

| Key | Value | Notes |
| --- | ----- | ----- |
| profile/mode | Supported profiling modes: `cpu`, `wall`, `object` for stackprof. | Use `profile` in (1), and `mode` in (2). Vernier backend only supports `wall` and `retained` at present time|
| async | Upload profile in a background thread. When this is set, profile redirect headers are not present in the response.
| interval | Sampling interval in microseconds. | |
| ignore_gc | Ignore garbage collection frames | |
| autoredirect | Redirect request automatically to Speedscope's page after profiling. | |
| context | Directory within the specified bucket in the selected storage where raw profile data should be written. | Only supported in (2). Defaults to `Rails.env` if not specified. |

> | backend | Profiler to use, either `stackprof` or `vernier`. Defaults to stackprof. Available only vernier is installed in the app, and Ruby version is at least 3.2.1 |

Reviewer: ```suggestion
| backend | Profiler to use, either `stackprof` or `vernier`. Defaults to stackprof. Vernier is only available when added separately in the application Gemfile, and the Ruby version is at least 3.2.1 |
```

Rails.application.config.app_profiler.middleware_action = AppProfiler::Middleware::UploadAction
```

## Profiler backends

It is possible to configure app_profiler to use the [`vernier`](https://github.com/jhawthorn/vernier) or stackprof. To use vernier, it must be added separately in the application Gemfile.

> The backend can be selected at dynamicall runtime using the `backend` parameter. The default backend to use when this parameter is not specified can be configured with:

Reviewer: ```suggestion
The backend can be selected dynamically at runtime using the `backend` parameter. The default backend to use when this parameter is not specified can be configured with:
```

```ruby
AppProfiler.profiler_backend = AppProfiler::StackprofBackend # or AppProfiler::VernierBackend
# OR
Rails.application.config.app_profiler.profiler_backend = AppProfiler::StackprofBackend # or AppProfiler::VernierBackend
```

By default, the stackprof backend will be used.

> In local development, changing the backend will change whether the profile is viewed in speedscope, or firefox-profiler.

Author: NIT move this to viewer PR

> In local development, changing the backend will change whether the profile is viewed in speedscope, or firefox-profiler.

Reviewer: ```suggestion
In local development, changing the backend will change whether the profile is viewed in speedscope or firefox-profiler.
```
